cli/azd/internal/scaffold/resource_meta.go

```diff
@@ -25,6 +25,39 @@ type ResourceMeta struct {
 	//
 	// To evaluate the actual values, see [Eval].
 	Variables map[string]string
+
+	// RoleAssignments are related role assignments the resource.
+	RoleAssignments RoleAssignments
+}
+
+type RoleAssignments struct {
+	Read  []RoleAssignment
+	Write []RoleAssignment
```

Comment:
How is the Read/Write distinction going to be used?

Reply:
It's unused today and just serves as metadata.

I suspect at some point, we may want the following:

```yaml
resources:
  api:
    uses:
    - name: storage
      permissions: [read | write]
```

To configure the level of permissioning required. I don't know what this means for the local user access, but this all came from @wbreza 's recent astute observation about blob storage access, that it's likely in some scenarios, the api may only want to serve files and not write content.